### PR TITLE
Fix #262 - allow hyphen in the username field

### LINKED ISSUE #262: Users can't migrate with usernames with non-alpha-numeric characters
screenshot 2015-04-20 13 51 00

@alicoding @cadecairos @thisandagain might be a problem with only accepting alpha-numeric characters. At the very least, people can't migrate. What should they do?

### PR DESCRIPTION
**Pull request by alicoding**
fixes #262

### CODE REVIEW

**Comment by alicoding** [comment]
@secretrobotron R?

File: templates/components/form/form.jsx [hunk 1]
```diff
@@ -78,6 +78,10 @@ var Form = React.createClass({
   dirty: function(id, origin) {
     return (err, valid) => {
       if(err) {
+        if(id === 'username' && this.state[id]) {
+          this.formError({field: 'username', message: 'Username must be between 1-20 characters, and may only include \"-\" and alphanumeric symbols.'});
```
Comment (secretrobotron) [inline templates/components/form/form.jsx]: `Must be between 1-20 characters, and use only \"-\" and alphanumeric symbols.`

Comment (secretrobotron) [inline templates/components/form/form.jsx]: `Must be 1-20 characters long and use only "-" and alphanumeric symbols.`

**Comment by secretrobotron** [comment]
r+

**Pull request merged**